pkg/test/framework/components/stackdriver/google.go

```diff
@@ -0,0 +1,213 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stackdriver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	cloudtrace "google.golang.org/api/cloudtrace/v1"
+	logging "google.golang.org/api/logging/v2"
+	monitoring "google.golang.org/api/monitoring/v3"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/genproto/googleapis/api/monitoredres"
+	cloudtracepb "google.golang.org/genproto/googleapis/devtools/cloudtrace/v1"
+	ltype "google.golang.org/genproto/googleapis/logging/type"
+	loggingpb "google.golang.org/genproto/googleapis/logging/v2"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+
+	md "istio.io/istio/pkg/bootstrap/platform"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+type realStackdriver struct {
+	monitoringService *monitoring.Service
+	loggingService    *logging.Service
+	traceService      *cloudtrace.Service
+	gcpEnv            md.Environment
+	projectID         string
+}
+
+type timeseriesQuery struct {
+	metricName   string
+	resourceType string
+}
+
+var (
+	_                 Instance = &realStackdriver{}
+	timeseriesQueries          = []timeseriesQuery{
```

Comment:
Can we include `gce_instance` queries here?

Reply:
Done.